src/cli/cli_dataset.cpp

```diff
@@ -355,5 +356,205 @@ ThreadError Dataset::ProcessPendingTimestamp(int argc, char *argv[])
+        {
+            VerifyOrExit(index < argc, error = kThreadError_Parse);
+            dataset.mIsMasterKeySet = true;
+            VerifyOrExit((length = Interpreter::Hex2Bin(argv[++index], dataset.mMasterKey.m8,
+                                                        sizeof(dataset.mMasterKey.m8))) == OT_MASTER_KEY_SIZE, error = kThreadError_Parse);
+            length = 0;
+        }
+        else if (strcmp(argv[index], "networkname") == 0)
+        {
+            VerifyOrExit(index < argc, error = kThreadError_Parse);
+            dataset.mIsNetworkNameSet = true;
+            VerifyOrExit((length = static_cast<int>(strlen(argv[++index]))) <= OT_NETWORK_NAME_MAX_SIZE,
+                         error = kThreadError_Parse);
+            memset(&dataset.mNetworkName, 0, sizeof(sDataset.mNetworkName));
+            memcpy(dataset.mNetworkName.m8, argv[0], static_cast<size_t>(length));
+            length = 0;
+        }
+        else if (strcmp(argv[index], "extpanid") == 0)
+        {
+            VerifyOrExit(index < argc, error = kThreadError_Parse);
+            dataset.mIsExtendedPanIdSet = true;
+            VerifyOrExit(Interpreter::Hex2Bin(argv[++index], dataset.mExtendedPanId.m8,
+                                              sizeof(dataset.mExtendedPanId.m8)) >= 0, error = kThreadError_Parse);
+        }
+        else if (strcmp(argv[index], "localprefix") == 0)
+        {
+            VerifyOrExit(index < argc, error = kThreadError_Parse);
+            dataset.mIsMeshLocalPrefixSet = true;
+            SuccessOrExit(error = otIp6AddressFromString(argv[++index], &prefix));
+            memcpy(dataset.mMeshLocalPrefix.m8, prefix.mFields.m8, sizeof(dataset.mMeshLocalPrefix.m8));
+        }
+        else if (strcmp(argv[index], "delaytimer") == 0)
+        {
+            VerifyOrExit(index < argc, error = kThreadError_Parse);
+            dataset.mIsDelaySet = true;
+            SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
+            dataset.mDelay = static_cast<uint32_t>(value);
+        }
+        else if (strcmp(argv[index], "panid") == 0)
+        {
+            VerifyOrExit(index < argc, error = kThreadError_Parse);
+            dataset.mIsPanIdSet = true;
+            SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
+            dataset.mPanId = static_cast<otPanId>(value);
+        }
+        else if (strcmp(argv[index], "channel") == 0)
+        {
+            VerifyOrExit(index < argc, error = kThreadError_Parse);
+            dataset.mIsChannelSet = true;
+            SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
+            dataset.mChannel = static_cast<uint16_t>(value);
+        }
+        else if (strcmp(argv[index], "binary") == 0)
+        {
+            VerifyOrExit((index + 1) < argc, error = kThreadError_Parse);
+            SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
```

Comment:
If I understand correctly, `length` is the length of the following argument in binary encoding.  If that is the case, why specify a `length` when it can just be determined from the length of the hex-encoded string?

In the example above, you have: `binary 3 820155`.  Why not just have: `binary 820155`?